internal/pkg/docker/dockerfile/lex.go

```diff
@@ -0,0 +1,241 @@
+	if err != nil {
+		lex.emitErr(fmt.Errorf("write '%s' to arguments buffer: %w", clean, err))
+		return nil
+	}
+
+	if hasLineContinuationMarker(lex.curLine) {
+		return lex.consumeInstr(name)
+	}
+	lex.emit(name)
+	return lexContent
+}
+
+// run walks through the state machine for the lexer.
+func (lex *lexer) run() {
+	for state := lexContent; state != nil; {
+		state = state(lex)
+	}
+	close(lex.instructions)
+}
+
+// stateFn represents a state machine transition of the scanner going from one INSTRUCTION to the next.
+type stateFn func(*lexer) stateFn
+
+// lexContent scans until we reach the end of the Dockerfile.
+func lexContent(l *lexer) stateFn {
+	isEOF, err := l.readLine()
+	if err != nil {
+		l.emitErr(err)
+		return nil
+	}
+	if isEOF {
+		l.emit(instrEOF)
+		return nil
+	}
+	line := strings.ToLower(strings.TrimLeftFunc(l.curLine, unicode.IsSpace))
+	switch {
+	case strings.HasPrefix(line, exposeInstrMarker):
+		return lexExpose
+	case strings.HasPrefix(line, heathCheckInstrMarker):
+		return lexHealthCheck
+	default:
+		return lexContent // Ignore all the other instructions, consume the line without emitting any instructions.
+	}
+}
+
+// lexExpose collects the arguments for an EXPOSE instruction and then emits it.
+func lexExpose(l *lexer) stateFn {
+	return lexInstruction(l, instrExpose)
+}
+
+// lexHealthCheck collects the arguments for a HEALTHCHECK instruction and then emits it.
+func lexHealthCheck(l *lexer) stateFn {
+	return lexInstruction(l, instrHealthCheck)
+}
+
+// lexInstruction collects all the arguments for the named instruction and then emits it.
+func lexInstruction(l *lexer, name instructionName) stateFn {
+	args := trimContinuationLineMarker(trimInstruction(l.curLine, instrMarkers[name]))
+	_, err := l.curArgs.WriteString(args)
+	if err != nil {
+		l.emitErr(fmt.Errorf("write '%s' to arguments buffer: %w", args, err))
+		return nil
+	}
+
+	if hasLineContinuationMarker(l.curLine) {
+		return l.consumeInstr(name)
+	}
+	l.emit(name)
+	return lexContent
+}
+
+// hasLineContinuationMarker returns true if the line wraps to the next line.
+func hasLineContinuationMarker(line string) bool {
+	for _, marker := range lineContinuationMarkers {
+		if strings.HasSuffix(line, marker) {
+			return true
+		}
+	}
+	return false
+}
+
+// trimInstruction trims the instrMarker prefix from line and returns it.
+func trimInstruction(line, instrMarker string) string {
+	normalized := strings.ToLower(line)
+	idx := strings.Index(normalized, instrMarker) + len(instrMarker)
+	return line[idx:]
+}
+
+// trimContinuationLineMarker returns the line without any continuation line markers.
+// If the line doesn't have a continuation marker, then returns it as is.
+func trimContinuationLineMarker(line string) string {
+	for _, marker := range lineContinuationMarkers {
+		if strings.HasSuffix(line, marker) {
+			return line[:len(line)-1-len(marker)]
```

Comment:
Is the extra `-1` for space `" "`?

Reply:
omg that was a bug! The test cases passed because all my line-wraps had an extra space `abc \` instead of `abc\`.
I've added a new test case and removed the -1